openedx/core/lib/xblock_serializer/block_serializer.py

```diff
@@ -112,6 +112,9 @@ def _serialize_html_block(self, block) -> etree.Element:
             olx_node.attrib["editor"] = block.editor
         if block.use_latex_compiler:
             olx_node.attrib["use_latex_compiler"] = "true"
+
+        # Escape any CDATA special chars
+        block.data = block.data.replace("]]>", "]]&gt;")
```

Comment:
I don't think this function should modify `block.data`, as that is an unexpected side effect. Would you mind updating this to either assign to a new variable or just put the `.replace()` call into the line below?

Reply:
Makes sense, just updated it.